### PR TITLE
Update CI jobs

### PR DESCRIPTION
**Pull request by Kichura**
1. Updates checkout, setup-java and setup-gradle to v4,
2. Replaces the old Gradle job *(`gradle/gradle-build-action`)* with new variant, *(`gradle/actions/setup-gradle`)*
~~3. Replaces the old validation job *(`gradle/wrapper-validation-action`)* with new variant. *(`gradle/actions/wrapper-validation`)* Removes the validation job in favor of setup-gradle,~~
4. Sets `persist-credentials` to false to prevent exposing to greater risks than imagined.

### CODE REVIEW

**Comment by jpenilla** [comment]
wrapper-validation action is no longer needed, setup-gradle v4 validates the wrapper

**Comment by Kichura** [comment]
> wrapper-validation action is no longer needed, setup-gradle v4 validates the wrapper

~~will remove.~~ removed.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,13 +12,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Git Repository
-        uses: actions/checkout@v3
-      - name: Validate Gradle wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: actions/checkout@v4
       - name: Setup Gradle
-        uses: gradle/gradle-build-action@v2
+        uses: gradle/actions/setup-gradle@v3
```
Comment (jpenilla) [inline .github/workflows/build.yml]: v4

Comment (Kichura) [reply .github/workflows/build.yml]: Skipped this earlier cuz origin repo did not reflect the v4 update in readme.

**Comment by kennytv** [comment]
Sorry about that - can you reopen a PR targeting the "new" master branch?

**Comment by Kichura** [comment]
> Sorry about that - can you reopen a PR targeting the "new" master branch?

will be reopened under new branch.

**Pull request closed without merging**